Source/Core/AST/AbsyCmd.cs

```diff
@@ -1352,14 +1350,19 @@ public Cmd(IToken /*!*/ tok)
     }
 
     public abstract void Emit(TokenTextWriter /*!*/ stream, int level);
-    public abstract void AddAssignedVariables(List<Variable> /*!*/ vars);
+
+    public IEnumerable<Variable> GetAssignedVariables() {
+      List<IdentifierExpr> ids = new();
+      AddAssignedIdentifiers(ids);
+      return ids.Select(id => id.Decl);
```

Comment:
What you are saying is that `.Decl` is available only after resolution, whereas the id is available after parsing, correct? If that is the case, could you just add a comment to this function that says something like "Can be called only after resolution" ? Or I'm mistaken.

Reply:
That's correct.